avalancheup-aws/src/apply/mod.rs

```diff
@@ -67,25 +117,418 @@ pub fn command() -> Command {
                 .required(false)
                 .num_args(0),
         )
+        .arg(
+            Arg::new("S3_REGION")
+                .long("s3-region")
+                .help("Sets the AWS S3 region")
+                .required(true)
+                .num_args(1)
+                .default_value("us-west-2"),
+        )
+        .arg(
+            Arg::new("S3_BUCKET")
+                .long("s3-bucket")
+                .help("Sets the S3 bucket")
+                .required(true)
+                .num_args(1),
+        )
+        .arg(
+            Arg::new("S3_KEY_PREFIX")
+                .long("s3-key-prefix")
+                .help("Sets the S3 key prefix for all artifacts")
+                .required(true)
+                .num_args(1),
+        )
+        .arg(
+            Arg::new("S3_UPLOAD_TIMEOUT")
+                .long("s3-upload-timeout")
+                .help("Sets the S3 upload timeout in seconds (default 30)")
+                .required(false)
+                .num_args(1)
+                .value_parser(value_parser!(u64))
+                .default_value("30")
+        )
+        .arg(
+            Arg::new("CHAIN_RPC_URL")
+                .long("chain-rpc-url")
+                .help("Sets the P-chain or Avalanche RPC endpoint")
+                .required(true)
+                .num_args(1),
```

Comment:
How would a user know the RPC URL ahead of time, if its created in the `apply` step?

Reply:
Yeah we only need a flag for spec file and everything else should be either in spec yaml file or default-spec flag